sdk/core/azure-core-amqp/inc/azure/core/amqp/connection.hpp

```diff
@@ -0,0 +1,173 @@
+    {
+      /** @brief Called when the connection state changes.
+       *
+       * @param connection The connection object whose state changed.
+       * @param newState The new state of the connection.
+       * @param oldState The previous state of the connection.
+       */
+      virtual void OnConnectionStateChanged(
+          Connection const& connection,
+          ConnectionState newState,
+          ConnectionState oldState)
+          = 0;
+
+      /** @brief Called when a new endpoint connects to the connection.
+       *
+       * @param connection The connection object.
+       * @param endpoint The endpoint that connected.
+       * @return true if the endpoint was accepted, false otherwise.
+       *
+       * @remarks Note that this function should only be overriden if the application is listening
+       * on the connection.
+       */
+      virtual bool OnNewEndpoint(Connection const& connection, Endpoint& endpoint)
+      {
+        (void)connection;
+        (void)endpoint;
+        return false;
+      }
+
+      /** @brief called when an I/O error has occurred on the connection.
+       *
+       * @param connection The connection object.
+       */
+      virtual void OnIoError(Connection const& connection) = 0;
+    };
+
+    class Connection final {
+    public:
+      Connection(
+          std::shared_ptr<Azure::Core::Amqp::Network::_internal::Transport> transport,
+          ConnectionOptions const& options,
+          ConnectionEvents* eventHandler = nullptr);
+
+      Connection(
+          std::string const& requestUri,
+          ConnectionOptions const& options,
+          ConnectionEvents* eventHandler = nullptr);
+
+      ~Connection();
+
+      Connection(Connection const&) = default;
+      Connection& operator=(Connection const&) = default;
+      Connection(Connection&&) noexcept = default;
+      Connection& operator=(Connection&&) = default;
+      Connection(std::shared_ptr<Azure::Core::Amqp::_detail::ConnectionImpl> impl) : m_impl{impl} {}
+
+      std::shared_ptr<Azure::Core::Amqp::_detail::ConnectionImpl> GetImpl() const { return m_impl; }
+
+      void Open();
+      void Listen();
+
+      void Close(
+          std::string const& condition,
+          std::string const& description,
+          Azure::Core::Amqp::Models::AmqpValue info);
+
+      void Poll() const;
+
+      uint32_t GetMaxFrameSize() const;
+      uint32_t GetRemoteMaxFrameSize() const;
+      void SetMaxFrameSize(uint32_t frameSize);
+      uint16_t GetMaxChannel() const;
+      void SetMaxChannel(uint16_t frameSize);
+      std::chrono::milliseconds GetIdleTimeout() const;
+      void SetIdleTimeout(std::chrono::milliseconds timeout);
+      void SetRemoteIdleTimeoutEmptyFrameSendRatio(double idleTimeoutEmptyFrameSendRatio);
+
+      void SetProperties(Azure::Core::Amqp::Models::AmqpValue properties);
+      Azure::Core::Amqp::Models::AmqpValue GetProperties() const;
+      uint64_t HandleDeadlines(); // ???
```

Comment:
Can you elaborate on the `// ???` comment, or add a TODO?

Reply:
I don't know what this function does, thus the ???